### PR TITLE
server: increase span config buffer sizes from 1MB to 4MB

### PR DESCRIPTION
**Pull request by erikgrinaker**
This patch increases the span config buffer sizes from 1MB to 4MB.  This is used to buffer span config rangefeed updates above the rangefeed frontier, until the frontier advances and buffered entries are flushed.

The rangefeed closed timestamp interval was recently increased from 200ms to 3s, with a 3s closed timestamp target duration, which can double the amount of buffered data. We have seen this buffer overflow in tests, requiring a rangefeed restart.

Touches #111195.
Epic: none
Release note: None

### CODE REVIEW

File: pkg/server/server.go [hunk 1]
```diff
@@ -776,7 +776,7 @@ func NewServer(cfg Config, stopper *stop.Stopper) (serverctl.ServerStartupInterf
 			clock,
 			rangeFeedFactory,
 			keys.SpanConfigurationsTableID,
-			1<<20, /* 1 MB */
+			4<<20, /* 4 MB */
```
Comment (stevendanna) [inline pkg/server/server.go:779]: @msbutler We should consider something similar for the span configuration streamer

Comment (stevendanna) [reply pkg/server/server.go:779]: Actually, over there we use the rangefeedcache whose buffer is expressed in the number of rows, so we actually have a very large buffer currently.

**Comment by erikgrinaker** [comment]
I see @arulajmani is OOO this week. Maybe you can have a look at this before the branch cut @andrewbaptist, since you're in this area anyway?

**Review by pav-kv — APPROVED** [review]

**Comment by pavelkalinnikov** [comment]
Is this a "global" buffer, or a buffer per *something*? I.e. what's the overall overhead?

**Comment by erikgrinaker** [comment]
It's global for the server.

**Review by arulajmani — COMMENTED** [review]
Change looks good. 

nit: Btw, the buffer is per store, not per server.

> Reviewed 1 of 1 files at r1, 2 of 2 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/111737#-:-NgtVwIr1TEhXvaQTmVr:b-ji974i)* status: :shipit: complete! 0 of 0 LGTMs obtained

**Pull request merged**